### PR TITLE
Iterative scheme for Task processing 

### PR DESCRIPTION
**Pull request by wkrzemien**
Introduces the iterative scheme:
User can declare number of iteration that given JPetUserTask should perform or even the infinite loop is acceptable with the special condtion that will stop the infite loop.
The condition is just a special boolean option ("stopIteration_bool")  that can be set in the JPetUserTask terminate method, and if it is set to true, then, the iteration will be interrupted.

In addition:
-Add make_unique implementation missing in C++11
-.Refactoring of JPetTaskIO:  separate JPetOutputHandler and JPetTaskIOTools. Also, add option that in case of empty outputFileType string passed to the constructor, the task will not write anything to the output.
-Also, some changes of JPetScopeLoader
-Create JPetTaskFactory and remove the Task creation logic from JPetManager
-Add bunch of unique pointers.
-Cleanup and simplification of JPetManager
-Change the way JPetManager returns error, now it throws exceptions that should be catched by the main functions.
-Remove JPetParamInterface and simplify large parts of coe
-Add bunch of tests

### CODE REVIEW

**Comment by alekgajos** [comment]
Please grep for "c-string" in the comments describing function parameters and update these comments as most of these params are no longer c-strings.

File: Core/JPetTaskIO/JPetTaskIO.cpp [hunk 1]
```diff
@@ -13,290 +13,178 @@
...
+  for (const auto& pTask : fSubTasks) {
+    pTask->init(fParams);
+
+    if (isInput()) {
+      auto firstEvent = 0ll;
+      auto lastEvent = 0ll;
+      bool isOK = false;
+      std::tie(isOK, firstEvent, lastEvent) = fInputHandler->getEventRange(fParams.getOptions());
+      if (!isOK) {
+        ERROR("Some error occured in getEventRange");
```
[the remark this answers is not in the document]
Comment (wkrzemien) [reply Core/JPetTaskIO/JPetTaskIO.cpp]: Ok. I am going to add it.

**Comment by wkrzemien** [comment]
Also, think about the case, where several modules have stop condition setup (naming clash)

File: Core/JPetTaskIO/JPetTaskIO.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (wkrzemien) [reply Core/JPetTaskIO/JPetTaskIO.cpp]: Added

**Comment by wkrzemien** [comment]
I fixed the comments.

**Pull request merged**